smrf/data/load_data.py

```diff
@@ -0,0 +1,184 @@
+import logging
+
+import numpy as np
+import utm
+
+from smrf.data import InputCSV, InputGribHRRR, InputNetcdf, InputWRF
+
+
+class InputData():
+    """
+    Class for loading and storing the data, either from
+    - CSV file
+    - HRRR grib files
+    - WRF
+    - Generic gridded NetCDF
+
+
+    The data will be loaded into a Pandas dataframe
+
+    """
+
+    VARIABLES = [
+        'air_temp',
+        'vapor_pressure',
+        'precip',
+        'wind_speed',
+        'wind_direction',
+        'cloud_factor',
+        'thermal',
+        'metadata'
+    ]
+
+    # Data variables and which module they belong to
+    MODULE_VARIABLES = {
+        'air_temp': 'air_temp',
+        'vapor_pressure': 'vapor_pressure',
+        'precip': 'precipitation',
+        'wind_speed': 'wind',
+        'wind_direction': 'wind',
+        'cloud_factor': 'cloud_factor',
+        'thermal': 'thermal'
+    }
+
+    DATA_FUNCTIONS = {
+        InputCSV.DATA_TYPE: InputCSV,
+        InputWRF.DATA_TYPE: InputWRF,
+        InputNetcdf.DATA_TYPE: InputNetcdf,
+        InputGribHRRR.DATA_TYPE: InputGribHRRR
+    }
+
+    # degree offset for a buffer around the model domain in degrees
+    OFFSET = 0.1
+
+    def __init__(self, smrf_config, start_date, end_date, topo):
+
+        self.smrf_config = smrf_config
+        self.start_date = start_date
+        self.end_date = end_date
+        self.time_zone = start_date.tzinfo
+        self.topo = topo
+
+        self._logger = logging.getLogger(__name__)
+
+        # get the buffer gridded data domain extents in lat long
+        self.model_domain_grid()
+
+        if 'csv' in self.smrf_config:
+            self.data_type = 'csv'
```

Comment:
Would it make sense to use the ` InputCSV.DATA_TYPE` here too?

Reply:
My thoughts would be that the first `csv` refers to the config option, but the second could use it.